extension/src/componentViewerProvider.ts

```diff
@@ -1,20 +1,128 @@
 import * as vscode from "vscode";
 import * as path from "path";
 import * as fs from "fs";
-import { ComponentModelProvider, ComponentRawModel } from "./componentModelProvider";
+import { parseStringSync } from "xml2js-parser";
+import { to_json as toJson, to_xml as toXml } from "xmljson";
+import * as promisify from "es6-promisify";
+import { ComponentModelProvider, ComponentRawModel, cxmlExtension } from "./componentModelProvider";
+import * as formattor from "formattor";
+
+interface TransitionParameter {
+    name: string;
+    triggeringEvent: string;
+    component: string;
+    stateSource: string;
+    stateMahineSource: string;
+    stateTarget: string;
+    stateMahineTarget: string;
+}
 
 export class ComponentViewerProvider implements vscode.TextDocumentContentProvider {
 
     private _onDidChange = new vscode.EventEmitter<vscode.Uri>();
     private context: vscode.ExtensionContext;
+    private transitionToCreate: TransitionParameter | undefined;
+    private uri: vscode.Uri;
 
-    public constructor(context: vscode.ExtensionContext) {
+    public constructor(context: vscode.ExtensionContext, uri: vscode.Uri) {
         this.context = context;
+        this.uri = uri;
+    }
+
+    public createTransition(args: Array<string>) {
+        this.transitionToCreate = {
```

Comment:
transitionToCreate is used in a promise (line 79). It means that you can have side effects (concurrent transition creations). So you cannot just use an instance variable...